docs/README.md

```diff
@@ -1,3 +1,201 @@
+
+### Primitive types
+
+The table below lists the Swift types and their corresponding C# types.
+
+| Swift Type                      | C# Type  |
+| ------------------------------- | -------- |
+| `Swift.Int64`                   | `long`   |
+| `Swift.UInt64`                  | `ulong`  |
+| `Swift.Int32`                   | `int`    |
+| `Swift.UInt32`                  | `uint`   |
+| `Swift.Int16`                   | `short`  |
+| `Swift.UInt16`                  | `ushort` |
+| `Swift.Int8`                    | `sbyte`  |
+| `Swift.UInt8`                   | `byte`   |
+| `Swift.UnsafeRawPointer`        | `IntPtr` |
+| `Swift.UnsafeMutableRawPointer` | `IntPtr` |
+| `Int`                           | `nint`   |
+| `UInt`                          | `nuint`  |
+| `Bool`                          | `bool`   |
+| `Float`                         | `float`  |
+| `Double`                        | `double` |
+
+Swift primitive types are implemented as frozen structs that conform to Swift-specific lowering processes handled by the runtime. However, such mappingcan fit within the underlying calling convention as these types are below the size limit for being passed by reference. 
+
+### Structs
+
+Swift structs are projected into C# as `IDisposable` classes, implementing `ISwiftStruct`, which inherit from `ISwiftNominalType` and `ISwiftDisposable`. This approach is chosen due to the semantic differences in lifecycle management between Swift and C# types. Bindings are generated with `SwiftStructAttribute` that inherits `SwiftNominalTypeAttribute`. This attribute contains information about the Swift type.
+
+Given the following Swift struct declaration:
+```swift
+    public struct BarInt {
+        public var X:Int; 
+        public init(x:Int) {
+            X = x;
+        }
+    }
+```
+
+The projection tooling will generate the following C#, with function bodies left empty for simplicity:
+```csharp
+    using System;
+    using System.Runtime.InteropServices;
+    using SwiftRuntimeLibrary;
+    using SwiftRuntimeLibrary.SwiftMarshal;
+    
+    namespace NewClassCompilerTests
+    {
+        [SwiftStruct("libNewClassCompilerTests.dylib",
+            "$s21NewClassCompilerTests6BarIntVMn", 
+            "$s21NewClassCompilerTests6BarIntVN", "")]
+        public class BarInt : ISwiftStruct
+        {
+            public BarInt(nint x)
+            {
+            }
+            internal BarInt(SwiftNominalCtorArgument unused)
+            {
+            }
+            public static SwiftMetatype GetSwiftMetatype()
```

Comment:
Do we have another doc describing the bases of the type system mapping? All these `ISwiftNominalType` and so on, along with for example this method and what are the expectations.

For this specific method, I'm curious what it's for. And if it's part of some contract, maybe it should be a static interface method?

Reply:
Here is the existing docs: https://github.com/xamarin/binding-tools-for-swift/blob/main/docs/Runtime-Public-API.md

I suggest to reduce the scope of the public API needed at this stage and migrate only the subset to this repository. Also, creating a contract (ABI) interface is a good idea, adding it to the list of tasks.